v0/destinations/yahoo_dsp/util.js

```diff
@@ -0,0 +1,283 @@
+const includes = {};
+const excludes = {};
+let seedList = [];
+let outputPayload = {};
+
+
+/**
+ * This function is used to check if there is any common value in keys and poiLocationType array. This is done to check
+ * if any of the required location type is present in the input or not. 
+ * @param {*} keys
+ * @param {*} poiLocationType
+ * @returns 
+ */
+function isCommonElement(keys, poiLocationType) {
+  return keys.some(item => poiLocationType.includes(item))
+}
+
+
+/**
+ *
+ * @param {*} audienceList  - It is the list of audience to be added in the form of array". eg.
+ * [
+ *   {"email": "abc@email.com"},
+ *   {"email": "abc@email.com"},
+ *   {"email": "abc@email.com"}
+ * ]
+ * @param {*} Config
+ * @returns The function returns an array of Audience List provided by the user like "email", "deviceId", "ipAddress".
+ * eg.[
+ * "251014dafc651f68edac7",
+ * "afbc34416ac6e7fbb9734",
+ * "42cbe7eebb412bbcd5b56",
+ * "379b4653a40878da7a584"
+ * ]
+ */
+const populateIdentifiers = (audienceList, Config) => {
+  const seedList = [];
+  const { audienceType } = Config;
+  const { hashRequired } = Config;
+  let listType;
+  if (isDefinedAndNotNullAndNotEmpty(audienceList)) {
+    // traversing through every userTraits in the add array for the traits to be added.
+    audienceList.forEach(userTraits => {
+      // storing keys of an object inside the add array.
+      const keys = Object.keys(userTraits);
+      // checking for the audience type the user wants to add is present in the input or not.
+      if(!keys.includes(audienceType)){
+        // throwing error if the audience type the user wants to add is not present in the input.
+        throw new CustomError(
+          `[Yahoo_DSP]:: Required property for ${audienceType} type audience is not available in an object`,
+          400
+        );
+      }
+      // here, hashing the data if is not hashed and pushing in the seedList array.
+      if (hashRequired) {
+        seedList.push(sha256(userTraits[audienceType]));
+      } else {
+        seedList.push(userTraits[audienceType]);
+      }
+    });
+  }
+  return seedList;
+};
+
+/**
+ * This function is used to create the output Payload.
+ * @param {*} audienceList - It is the list of audience to be added in the form of array". eg.
+ * [
+ *   {"email": "abc@email.com"},
+ *   {"email": "abc@email.com"},
+ *   {"email": "abc@email.com"}
+ * ]
+ * @param {*} Config 
+ * @returns 
+ */
+const createPayload = (audienceList, Config) => {
+  const accountId = Config.accountId;
+  let seedList = [];
+   // Populating Seed List that conains audience list to be updated
+  seedList = populateIdentifiers(audienceList, Config);
+  // throwing the error if nothing is present in the seedList
+  if (seedList.length === 0) {
+    throw new CustomError(
+      `[Yahoo_DSP]:: No attributes are present in the '${key}' property.`,
+      400
+    );
+  }
+  // Creating outputPayload
+  outputPayload = { ...outputPayload, accountId, seedList };
```

Comment:
This should be declared here itself to avoid dangling objects which can cause memory leak

Reply:
Do we need object spreading here?

Reply:
Updated it.